### DISCUSSION TITLE
Realistic ZF precoder

### DISCUSSION THREAD

**Discussion by Mio-TW** [NVlabs/sionna#665] [Q&A] [answered]
Hi Sionna team,

I tried the [example](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html), and in the "Downlink Transmission in the Frequency Domain" cases, the ZF precoder is applied to reduce the multi-path interference in MIMO scenarios.
However, in a realistic scenario, the transmitter does not have the perfect knowledge of the channel. 
As mention in the tutorial, it may take **outdated transmit CSI** into account for signal precoding.
I am curious about how to implement the outdated transmit CSI in the simulations? 
Do we need the construct the overall CSI feedback process structure? 
Or are there any simpler ways to obtain the outdated transmit CSI?
Thank you.

**Comment by jhoydis** [comment, marked as answer]
Hi @Mio-TW,

As a simple approximation, you could just add some noise to the perfect CSI.
Or you could generate 2*num_ofdm_symbols channel realizations, take the first for the computation of the precoder, and then simulate the downlink transmissions for the second half of the channel realizations. This would take the "outdated" nature specifically into account.
Otherwise, you would need to implement channel estimation yourself.

**Reply by Mio-TW** [reply]
Hi @jhoydis,

Thank you very much for your suggestions. 👍 
It's very helpful, and I will try it.
